npm/private/npm_translate_lock.bzl

```diff
@@ -366,6 +380,11 @@ def _gen_npm_imports(lockfile, attr):
             friendly_name not in attr.lifecycle_hooks_exclude
         )
 
+        lifecycle_hooks_env = []
+        for package in attr.lifecycle_hooks_env:
```

Comment:
should check against friendly_name and unfriendly_name variants as well, which allows the user to set the env for a particular package version such as `foo@1.2.3`. shortest version of this code block would look like

```
        lifecycle_hooks_env = attr. lifecycle_hooks_envs.get(name, [])[:]
        lifecycle_hooks_env.extend(attr.lifecycle_hooks_envs.get(friendly_name, []))
        if unfriendly_name:
            lifecycle_hooks_env.extend(attr.lifecycle_hooks_envs.get(unfriendly_name, []))
```

Reply:
Actually, this PR will make this better by providing a helper function: https://github.com/aspect-build/rules_js/pull/293

Reply:
Should land shortly so you can rebase and use that

Reply:
Landed

Reply:
Got it, because I also need to check whether the user define `"*": [key=value]`,
could I update the source to 
```
 lifecycle_hooks_env = []
        for package in attr.lifecycle_hooks_envs:
            if package == "*" or package == name or package == friendly_name or package == unfriendly_name:
                lifecycle_hooks_env.extend(package_lifecycle_hooks_env[package])
```

Reply:
You could do it in one line :)
```
lifecycle_hooks_env = _gather_values_from_matching_names(attr.lifecycle_hooks_envs, "*", name, friendly_name, unfriendly_name)
```